app/test_main.py

```diff
@@ -1 +1,53 @@
-# write your code here
+import pytest
+from unittest import mock
+
+
+from app.main import can_access_google_page
+
+
+@pytest.fixture()
+def mocked_valid_google_url() -> None:
+    with mock.patch("app.main.valid_google_url") as mocked_func:
+        yield mocked_func
+
+
+@pytest.fixture()
+def mocked_has_connection() -> None:
+    with mock.patch("app.main.has_internet_connection") as mocked_func:
+        yield mocked_func
+
+
+def test_valid_connection_and_url_exist(mocked_valid_google_url: any,
+                                        mocked_has_connection: any
+                                        ) -> None:
+    mocked_valid_google_url.return_value = True
+    mocked_has_connection.return_value = True
+    url = "http//:www.google.com/"
```

Comment:
Are you sure this "url" variables are necessary? They are never used outside function call, so maybe it can be simply removed?

Reply:
Yes, "url" is better be there, cause if someone would be needed to change url (in different tests, for example), they can change "url" variable. And this makes code easier to read) And there was a problem in similar case with some mentors (I should add a variable in that case, so I added it here too), so it's up to them: someone likes variables there, someone - don't)

Reply:
i see. allright then :)

Reply:
In this case, you can make your variable a constant, because you use the same value everywhere and it doesn't matter for your tests what url you use

Reply:
Ok - done.